tree/dataframe/inc/ROOT/RDF/RInterfaceBase.hxx

```diff
@@ -108,9 +108,7 @@ protected:
          const auto &retTypeID = typeid(typename RetType::value_type);
          const auto &colName = colNames[0]; // we have only one element in there
          const auto *define = fColRegister.GetDefine(colName);
-         const auto &expectedTypeID =
-            define ? define->GetTypeId() : RDFInternal::TypeName2TypeID(GetColumnType(colName));
-         if (retTypeID != expectedTypeID)
+         if (define ? (retTypeID != define->GetTypeId()) : (retTypeID != RDFInternal::TypeName2TypeID(GetColumnType(colName))))
```

Comment:
I find the previous code much more readable.  Could you paste here the compiler diagnostic from gcc13 to see if we can find a better fix?